noxfile.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+
+import nox
+from nox import Session, session
+
+python_versions = ['3.5', '3.6', '3.7', '3.8', '3.9', '3.10']
+nox.options.sessions = ['tests', 'docs', 'style']
+nox.options.reuse_existing_virtualenvs = True
+
+
+@session(python=python_versions)
+def tests(session: Session) -> None:
+    """Run the tests."""
+    args = session.posargs or ['--buffer']
+
+    session.install('--upgrade', 'pip', 'setuptools', 'wheel', 'coverage')
+    session.install('-e', '.')
+
+    try:
+        session.run(
+            'coverage', 'run', '--parallel', '--module', 'unittest', *args
+        )
+    finally:
+        if session.interactive:
+            session.notify('coverage', posargs=[])
+
+
+@session
+def coverage(session: Session) -> None:
+    """Produce the coverage report."""
+    args = session.posargs or ['report', '--show-missing']
+
+    session.install('coverage')
+
+    if not session.posargs and any(Path().glob('.coverage.*')):
+        session.run('coverage', 'combine')
+
+    session.run('coverage', *args)
+
+
+@session
+def docs(session: Session) -> None:
+    """Produce the coverage report."""
+    args = session.posargs or ['-b', 'html', 'doc/source', 'doc/build']
+
+    session.install(
+        'sphinx',
+        'sphinx-autodoc-typehints',
+    )
```

Comment:
please use doc/requirements.txt (adding sphinx in there) to avoid things going out of sync.

Reply:
Done.